aggregator/internal/pkg/aggregator.go

```diff
@@ -377,3 +387,44 @@ func (agg *Aggregator) AddNewTask(batchMerkleRoot [32]byte, senderAddress [20]by
 	agg.AggregatorConfig.BaseConfig.Logger.Info("- Unlocked Resources: Adding new task")
 	agg.logger.Info("New task added", "batchIndex", batchIndex, "batchIdentifierHash", "0x"+hex.EncodeToString(batchIdentifierHash[:]))
 }
+
+// long-lived gorouting that periodically checks and removes old Tasks from stored Maps
+func (agg *Aggregator) ClearTasksFromMaps(period time.Duration, blocksOld uint64) {
+	defer func() {
+		err := recover() //stops panics
+		if err != nil {
+			agg.logger.Error(err.(string))
+		}
+	}()
+
+	agg.AggregatorConfig.BaseConfig.Logger.Info(fmt.Sprintf("- Removing finalized Task Infos from Maps every %v", period))
+	lastIdxDeleted := uint32(0)
+
+	for {
+		time.Sleep(period)
```

Comment:
done

Reply:
Wait, no, actually we have to execute it at the beggining, because we have `continue` closures that want to break the current execution and wait for next execution